pincer/utils/shards.py

```diff
@@ -0,0 +1,14 @@
+# Copyright Pincer 2021-Present
+# Full MIT License can be found in `LICENSE` at the project root.
+
+def calculate_shard_id(guild_id, num_shards):
+    """Calculates the shard receiving the events for a specified guild
+
+    Parameters
+    ----------
+    guild_id : Optional[~pincer.utils.snowflake.Snowflake]
+        The guild_id of the shard to look for
+    num_shards : Optional[int]
+        The number of shards.
```

Comment:
Why are these typed as `Optional[TYPE]`?

Reply:
No guild id = get dm guild. This means that you can pass in ctx.guild_id if all interactions and it will get the correct guild if it is a dm.
No num_shards = default to most recent shard count. Just makes things easier to write.
I will update docs when i get home